drivers/serial/uart_nrfx_uarte.c

```diff
@@ -20,19 +41,73 @@
      defined(CONFIG_UART_2_INTERRUPT_DRIVEN)) || \
     (defined(CONFIG_UART_3_NRF_UARTE) &&         \
      defined(CONFIG_UART_3_INTERRUPT_DRIVEN))
-	#define UARTE_INTERRUPT_DRIVEN	(1u)
+	#define UARTE_INTERRUPT_DRIVEN	1
 #endif
 
+#define MS_PER_TICK (CONFIG_SYS_CLOCK_TICKS_PER_SEC <= 1000 ?	\
+		     1000 / CONFIG_SYS_CLOCK_TICKS_PER_SEC : 1)
+/*
+ * RX timeout is divided into time slabs, this define tells how many divisions
+ * should be made. More divisions - higher timeout accuracy and processor usage.
+ */
+#define RX_TIMEOUT_DIV 5
+
+#ifdef CONFIG_UART_ASYNC_API
+struct uarte_async_cb {
+	uart_callback_t user_callback;
+	void *user_data;
+
+	/* tx_buf and tx_size have to be volatile as they are used as busy flags
+	 * in uart_tx and uart_poll_out. If both tx_buf and tx_size is set then
+	 * there is currently ongoing asynchronous transmission. If only tx_size
+	 * is bigger than 0 and tx_buf is NULL, then there is ongoing
+	 * transmission by uart_poll_out
+	 */
+	const u8_t *volatile tx_buf;
+	volatile size_t tx_size;
+	struct k_timer tx_timeout_timer;
+
+	u8_t *rx_buf;
+	size_t rx_size;
```

Comment:
This field is not used, it is only assigned a value in a few places, and one of them is `rx_next_size` which in turn is never assigned any value (it stays initialized by default to `0`). Both these fields should be removed.